### PR TITLE
Resolves dmarcotte/easy-move-resize#38

### PR DESCRIPTION
**Pull request by patchon**
This commit does a few things, but mostly it adds the option to use the
middle mouse button for resizing (this behaviour is configurable in
X11).

It also fixes a few bugs and adds some improvement, namely,
- Bring Window to front is saved correctly, and menu updated
  accordingly.
- Disabled behaviour is now saved and not only ad hoc, ie. if you disable
  the application and exits, the application will be disabled on launch
  again.
- When choosing to disable the application, all the menu items are now
  properly disabled.
- Renamed a few things to fit the above.

Hope I followed your coding style, this is my first time opening Xcode and looking at Objective-C, so please go easy on me. 
But I think this is pretty straight forward and doesn't do any harm.

### CODE REVIEW

File: easy-move-resize/EMRAppDelegate.m [hunk 1]
```diff
@@ -353,19 +395,30 @@ - (IBAction)toggleBringWindowToFront:(id)sender {
...
+    BOOL newState = ![menu state];
+    [menu setState:newState];
+    [preferences setShouldMiddleClickResize:newState];
+}
+
 - (IBAction)toggleDisabled:(id)sender {
     EMRMoveResize* moveResize = [EMRMoveResize instance];
     if ([_disabledMenu state] == 0) {
         // We are enabled. Disable...
+        NSLog(@"disabling");
```
Comment (dmarcotte) [inline easy-move-resize/EMRAppDelegate.m:409]: This debug logging can probably be deleted?

File: easy-move-resize/EMRPreferences.h [hunk 2]
```diff
@@ -1,14 +1,16 @@
 // Preferences can alternativevly be managed from the Terminal:
 //   Read:
-//     `defaults write org.dmarcotte.Easy-Move-Resize ModifierFlags CMD,CTRL`
+//     `defaults read org.dmarcotte.Easy-Move-Resize ModifierFlags CMD,CTRL`
```
Comment (dmarcotte) [inline easy-move-resize/EMRPreferences.h:3]: Nice catch :)

File: easy-move-resize/EMRAppDelegate.m [hunk 3]
```diff
@@ -353,19 +395,30 @@ - (IBAction)toggleBringWindowToFront:(id)sender {
...
         [_disabledMenu setState:YES];
-        [self setMenusEnabled:YES];
+        [self setMenusEnabled:NO];
         [self disableRunLoopSource:moveResize];
+        [preferences setShouldBeDisabled:YES];
     }
     else {
         // We are disabled. Enable.
+        NSLog(@"dddSomething To Printaaaa");
```
Comment (dmarcotte) [inline easy-move-resize/EMRAppDelegate.m:417]: This should probably be deleted too?

**Review by dmarcotte — COMMENTED** [review]
Code's looking good!  I like the care put into naming and updating the docs too, thanks.  Just a few little things to sort out before we can merge:

- I added a couple of notes on what looks like some stray debug logging that should probably be deleted before merging
- Why did you feel like you had to commit all the build files?  Ideally we would not have to
- I'm not sure I agree with the design decision to persist the "Disabled" state.  If someone disables, then quits some time after... when they launch again, I think it's fair to assume they want the app enabled.  Then can quit or re-disable if that's not true.  Thoughts?

Hope that all makes sense, and thanks sending this along.

File: easy-move-resize/EMRAppDelegate.m (hunk 1 above)
Comment (patchon) [reply easy-move-resize/EMRAppDelegate.m:409]: Yes, definitely, sorry forgot obout that one.

File: easy-move-resize/EMRAppDelegate.m (hunk 3 above)
Comment (patchon) [reply easy-move-resize/EMRAppDelegate.m:417]: Yeps :)

File: easy-move-resize/EMRPreferences.h (hunk 2 above)
Comment (patchon) [reply easy-move-resize/EMRPreferences.h:3]: Not sure about why this does't work for me. I cant find the settings for the app anywhere (I dont know much about macos), but the settings are saved and stored correctly, I just cant figure out where. With theese commands, I don't get any results..

**Comment by patchon** [comment]
> Code's looking good! I like the care put into naming and updating the docs too, thanks. Just a few little things to sort out before we can merge:
> 
> * I added a couple of notes on what looks like some stray debug logging that should probably be deleted before merging
Yeps, my mistake .

> * Why did you feel like you had to commit all the build files?  Ideally we would not have to
Well, sorry about that, I was just a little bit to eager and wanted to share my build, thats all. Totally fine to remove. 

> * I'm not sure I agree with the design decision to persist the "Disabled" state.  If someone disables, then quits some time after... when they launch again, I think it's fair to assume they want the app enabled.  Then can quit or re-disable if that's not true.  Thoughts?
Haha, well I guess its more of a philosophical question, to be honest I'm not sure why you want to disable the app at all :) But hey, I guess the option is there for a good reason. The reason why I decided to save the state was because I like consistency, ie. I wanted all options behaviour to work the same - and while I was at it fixing the state for "Bringing Window to front", I thought I just might fix all of them to work the same :) 

But I don't put any pride in that, it's your app and I'm fine with whatever decision you make =) 

> Hope that all makes sense, and thanks sending this along.

Just tell me if you want me to update the PR, or if you'll fix it yourself. 

Again, thanks for an awesome app, this has made my transition to macOS pretty smooth :)

File: easy-move-resize/EMRPreferences.h (hunk 2 above)
Comment (dmarcotte) [reply easy-move-resize/EMRPreferences.h:3]: Ah, interesting... thanks for the heads up.  The key there must be out of date now.  I'll have a look later.

**Comment by dmarcotte** [comment]
Thanks, @patchon!  I'd love it if you could update the PR to get this over the finish line, so here's the info you need:
- let's revert those commits adding the build files (are you comfortable using git to edit them out of the history and then force pushing here?)---I'll build and make a release once you're merged
- let's stick with _not_ persisting `disabled` across restarts

Lemme know if I missed anything, and thanks again.  Really appreciate the contribution here.  

(And I'm really glad to hear this helped with your macOS transition :D Helping my transition is why it got built in the first place!)

**Comment by patchon** [comment]
[See this pull request instead](https://github.com/dmarcotte/easy-move-resize/pull/57).

**Pull request closed without merging**